From 1c4a466e18dbad07f86d95ec02f099c0da7c15fc Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Tue, 30 Mar 2021 11:10:24 +0300
Subject: [PATCH] [SW-0] preboot: remove weak alias

Remove weak alias usage, replace with weak function and function call.
Removed for readbility - functionality is not changed.

Change-Id: Id394e1fdd7943c9114e64c52745f55d4f25d7159
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/preboot/preboot.c b/preboot/preboot.c
index 5f8526b..91f59e9 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -234,8 +234,10 @@
 	return false;
 }
 
-bool preboot_is_valid_address_range(u64 addr, int len)
-	__attribute__((weak, alias("preboot_sram_is_valid_address_range")));
+__weak bool preboot_is_valid_address_range(u64 addr, int len)
+{
+	return preboot_sram_is_valid_address_range(addr, len);
+}
 
 __weak void preboot_dev_enable_caps(void) {}
 
